lib/src/jpegr.cpp

```diff
@@ -723,11 +727,43 @@ status_t JpegR::decodeJPEGR(jr_compressed_ptr jpegr_image_ptr, jr_uncompressed_p
     gainmap_image.height = jpeg_dec_obj_gm.getDecompressedImageHeight();
 
     if (gainmap_image_ptr != nullptr) {
-      gainmap_image_ptr->width = gainmap_image.width;
-      gainmap_image_ptr->height = gainmap_image.height;
       gainmap_image_ptr->pixelFormat = gainmap_image.pixelFormat;
-      memcpy(gainmap_image_ptr->data, gainmap_image.data,
-             gainmap_image_ptr->width * gainmap_image_ptr->height);
+      if (jpeg_dec_obj_yuv420.getDecompressedImageWidth() % gainmap_image.width == 0 &&
+          jpeg_dec_obj_yuv420.getDecompressedImageHeight() % gainmap_image.width == 0) {
+        gainmap_image_ptr->width = gainmap_image.width;
+        gainmap_image_ptr->height = gainmap_image.height;
+
+        memcpy(gainmap_image_ptr->data, gainmap_image.data,
```

Comment:
Actually parent CL has an issue. For multi channel gain maps, the api has to allocate memory for 3 components for gainmap image. Currently api is always allocating gainmap buffer size for grayscale only. This needs to be corrected.